build.sbt

```diff
@@ -0,0 +1,171 @@
+organization in ThisBuild := "io.gloriousfuture"
+organizationName in ThisBuild := "The Glorious Future"
+
+version in ThisBuild := "0.0.1"
+scalaVersion := "2.11.8"
+
+licenses in ThisBuild += ("Apache-2.0", url("http://opensource.org/licenses/apache-2.0"))
+
+lazy val root = Project("genrpc-root", file("."))
+  .aggregate(
+    // RPC client code generators
+    `genrpc-core`,
+    // TODO: Move these projects out into a "genspec" repo
+    // OpenAPI spec generators
+    `genspec-openapi-core`,
+    `genspec-openapi-circe`,
+    `genspec-openapi-play`,
+    `genspec-openapi-yaml`,
+    // SBT plugins
+    `genspec-sbt-plugin`,
+    `genspec-play-plugin`
+  )
+  .settings(
+    publish := {},
+    publishLocal := {}
+  )
+
+def commonProject(id: String, path: String): Project = {
+  Project(id, file(path))
+    .settings(
+
+      scalacOptions ++= {
+        // the deprecation:false flag is only supported by scala >= 2.11.3, but needed for scala >= 2.11.0 to avoid warnings
+        (CrossVersion.partialVersion(scalaVersion.value) match {
+
+          case Some((2, scalaMinor)) if scalaMinor >= 11 =>
+            // For scala versions >= 2.11.3
+            Seq("-deprecation:false", "-target:jvm-1.8", "-Ywarn-unused")
+
+          case Some((2, scalaMinor)) if scalaMinor < 11 =>
+            // For scala versions 2.10.x we can't use certain flags
+            Seq.empty
+
+        }) ++ Seq(
+          "-feature",
+          "-unchecked",
+          "-Xfatal-warnings",
+          "-Xfuture",
+          "-Ywarn-value-discard"
+        )
+      },
+
+      libraryDependencies ++= Seq(
+        // Test-only dependencies
+        Libraries.scalatest
+      ).map(_ % Test)
+    )
+}
+
+def sbtPluginProject(id: String, path: String): Project = {
+  commonProject(id, path).settings(
+    sbtPlugin := true,
+    sbtVersion := "0.13.13",
+    scalaVersion := "2.10.6"
+  )
+}
+
+def libProject(id: String, path: String): Project = {
+  commonProject(id, path).settings(
+    scalaVersion := "2.11.8"
+  )
+}
+
+// Client-side libraries
+
+lazy val `genrpc-core` = libProject("genrpc-core", "genrpc/core")
+  .settings(
+    name := "genrpc-core"
+  )
+
+// TODO: Move to scripted
+//lazy val examplePlayYaml = (project in file("example/play-yaml"))
```

Comment:
I want to do something like this for testing the sbt plugin: http://eed3si9n.com/testing-sbt-plugins